Update CDN Documentation (#1110)
zeit/docs: pages/docs/v2/network/caching.mdx

Static files are cached for 31 days. However, when you redeploy the cache is effectively invalidated so we always serve the latest version.

By default we return a `Cache-Control` header containing `public, max-age=0, must-revalidate` to prevent clients (e.g. browsers) from caching the file locally. This gives you the most flexibility as users get the latest file from our Global CDN immediately after deploying. This can be overridden with the [routes](https://zeit.co/docs/v2/deployments/routes/#headers) property in your `now.json` file

## Serverless Functions (Lambdas)

In order for our CDN to cache the response of a Lambda, you need to include the response header `Cache-Control` with **any** of the following directives:

- `s-maxage=N`
- `maxage=N, public`
- `maxage=N, immutable`

Where `N` is the number of seconds the response should be cached for. The response must also be cacheable as detailed above.

You can read more about the suggested way to define this property [here](/docs/v2/advanced/concepts/cdn-and-global-distribution/#smart-cdn).

## Stale-While-Revalidate

Our CDN supports a powerful recent extension to the `Cache-Control` header called `stale-while-revalidate`.

> The benefit of using `stale-while-revalidate` is that we can serve a resource from our CDN cache, but at the same time, update it **in the background** by calling your lambda.

Author: Applied 👍

Some situations where `stale-while-revalidate` is of great value:

> - Your content changes frequently but it takes a significant amount of time to regenerate. Maybe because of database or API access.

Author: Applied 👍

- Your content changes infrequently but you want to have the flexibility to update it (to fix a typo, for example) and don't wait for the cache to expire.

In both cases, we recommend using:

```
Cache-Control: s-maxage=1, stale-while-revalidate
```

<Caption>
  An example <InlineCode>Cache-Control</InlineCode> header that enables content
  to be updated in the background.
</Caption>

Which tells our CDN: serve from cache, but update it, if requested after 1 second.

When the CDN receives a request with `Pragma: no-cache` (such as when the browser devtools are open), it will revalidate any stale resource synchronously, instead of in the background.

## Cache Invalidation

Every deployment has a unique key used for caching based on the deployment url created at build time. This means that users will never see content from a previous deployment and there's normally no need to invalidate it.

> If you ever need to invalidate the CDN cache, you can always create a new deployment.

Reviewer: ```suggestion
If you ever need to invalidate the CDN cache, you can always re-deploy.
```
I know I wrote this but "create a new deployment" sounds like a lot of work compared to re-deploying

  lastEdited: '2019-08-12T12:18:55.000Z'
}

Our [Smart CDN](/smart-cdn) caches your content at the edge in order to serve data to your users as fast as possible.

## Cacheable Responses

Only responses with HTTP status `200` to a `GET` or `HEAD` request are cached by our CDN. Other status codes or HTTP methods are never cached. In addition to this, there are **various other circumstances where caching does not take place**:

- Responses that exceed `10MB` in content length.
- Requests that contain the `Range` header.
- Requests that contain the `Authorization` header.
- Requests that contain a `_now_no_cache=1` cookie.
- Requests that contain a `?_now_no_cache=1` request parameter.
- Responses that contain the `no-cache` directive in their `Cache-Control` header.

## Static Files

> Static files are **automatically cached at the edge** after the first request. There are no manual adjustments needed to make this work.

Reviewer 2: ```suggestion
Static files are **automatically cached at the edge** after the first request. There are no manual adjustments necessary to make this work.
```

Static files are cached for 31 days. However, when you redeploy the cache is effectively invalidated so we always serve the latest version.

By default we return a `Cache-Control` header containing `public, max-age=0, must-revalidate` to prevent clients (e.g. browsers) from caching the file locally. This gives you the most flexibility as users get the latest file from our Global CDN immediately after deploying. This can be overridden with the [routes](https://zeit.co/docs/v2/deployments/routes/#headers) property in your `now.json` file

## Serverless Functions (Lambdas)

> In order for our CDN to cache the response of a Lambda, you need to include the response header `Cache-Control` with **either** of the following directives:

Reviewer 2: ```suggestion
In order for our CDN to cache the response of a Lambda, you need to include the response header `Cache-Control` with **any** of the following directives:
```
